game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/BattleStep.java

```diff
@@ -0,0 +1,88 @@
+package games.strategy.triplea.delegate.battle.steps;
+
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.triplea.delegate.ExecutionStack;
+import games.strategy.triplea.delegate.IExecutable;
+import games.strategy.triplea.delegate.battle.BattleActions;
+import games.strategy.triplea.delegate.battle.BattleState;
+import java.util.List;
+import lombok.AllArgsConstructor;
+
+/**
+ * A step in a battle.
+ *
+ * <ol>
+ *   <ul>
+ *     Each step can have 0 or more names. These names are shown in the {@link
+ *     games.strategy.triplea.ui.BattlePanel}
+ *   </ul>
+ *   <ul>
+ *     Each step can also have an executable. See {@link BattleAtomic} on what an executable is
+ *     comprised of.
+ *   </ul>
+ * </ol>
+ */
+@AllArgsConstructor
+public abstract class BattleStep {
+
+  /**
+   * The current state of the battle
+   *
+   * <p>Note: This is current mutable so the underlying data can change.
+   */
+  protected final BattleState battleState;
+
+  /** Actions that can occur in a battle that require interaction with {@link IDelegateBridge} */
+  protected final BattleActions battleActions;
```

Comment:
Inheritance is a very strong form of coupling, using `protected` variables is also potentially very painful as well when considering inheritance hierarchies. In some cases as well some implementations may not need all of the protected variables.

What are your thoughts to convert this class to an interface? In such a case, it seems like we only need to move these two variables to the child classes. The level of code duplication is pretty negligable since they already have the all arg constructors and in turn each of the child classes would be much more strongly encapsulated. WDYT, any concerns with that suggestion?

Reply:
Before the conversion to use `MustFightBattle` as the `BattleState` there was more of a reason.  I can't think of a good reason to keep it inherited so I'll change it to an interface..